docs/rfcs/001-project-specific-configuration.md

```diff
@@ -0,0 +1,124 @@
+# Project-Specific Configuration
+
+## Status
+
+Proposed
+
+## Summary
+
+Add the ability to have project-specific settings that override the global settings in the `~/.atom/config.{cson,json}` file on a per-project basis.
+
+## Motivation
+
+People who use Atom often work on multiple projects. Sometimes these projects have very different requirements for code formatting, tools to use, or other features that will necessitate different configuration of the editing environment. Currently, if one needs different settings for different projects, one has to use a community package, use [editorconfig](http://editorconfig.org/) which has a limited set of configuration values, or resort to manually changing configuration each time one switches to a different project.
+
+## Explanation
+
+There are two configuration files: the global `~/.atom/config.cson` and the project-specific `atom-config.cson`. (Hereafter these two files will be referred to as the "global settings" and "project settings" respectively.) The project settings file is stored in the root directory of the project root.
+
+When reading from the configuration using `atom.config.get()`, the contents of the two configuration files is merged using [`Object.assign`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign) or some similar mechanism, with the values in the project settings taking precedence over the global settings. For example:
```

Comment:
Note that the example retrieval is from within the project, and maybe add an example for global retrieval?

Reply:
Added.